### ISSUE TITLE
Xcode 13 error builds

### ISSUE THREAD

**Issue by nriccar** [dotintent/react-native-ble-plx#899]
Hello,

I noticed my Xcode updated to 13 
![Screenshot 2021-09-22 at 16 07 46](https://user-images.githubusercontent.com/49196314/134359257-7f0f021b-7fb8-4850-ab39-96060b46b62f.png)

And right after that I started getting this error at build:

![Screenshot 2021-09-22 at 16 08 04](https://user-images.githubusercontent.com/49196314/134359305-f38ea58b-af6c-4f19-ab33-c4952e372502.png)
![Screenshot 2021-09-22 at 16 08 25](https://user-images.githubusercontent.com/49196314/134359356-31eb3b44-54c3-4f87-bb00-093fc5a64008.png)
![Screenshot 2021-09-22 at 16 08 32](https://user-images.githubusercontent.com/49196314/134359374-5caf287b-e532-48ec-8d18-be85980168c0.png)
![Screenshot 2021-09-22 at 16 08 40](https://user-images.githubusercontent.com/49196314/134359400-057b2977-192b-4312-b1d0-436d74e04718.png)
![Screenshot 2021-09-22 at 16 08 49](https://user-images.githubusercontent.com/49196314/134359425-6c1dd7aa-33c2-4171-b79d-8ee28f390515.png)

I run pod update 
```
Downloading dependencies
Installing MultiplatformBleAdapter 0.1.7 (was 0.1.6)
Installing react-native-ble-plx 2.0.2 (was 2.0.1)
Generating Pods project
Integrating client project
Pod installation complete! There are 81 dependencies from the Podfile and 71 total pods installed.
```

But same error 

Any guess?

**Comment by wildseansy** [comment]
Seems like they're working on a fix: https://github.com/dotintent/MultiPlatformBleAdapter/commit/669e438acb384eee74f3f1ada3ba259cf7b7965b

but appears it hasn't been released yet

**Comment by nriccar** [comment]
Thanks for the heads up. Ill check that one!

**Comment by xanderdeseyn** [comment]
Seeing this as well, any idea when they will release the fix?

**Comment by diogoviannaaraujo** [comment]
This has made impossible to develop on iOS 15 since XCode 12 does not accept iOS 15 devices as a development target.

**Comment by wildseansy** [comment]
Maybe it wasn't clear, but can be fixed while they prepare release through patch package

https://github.com/dotintent/MultiPlatformBleAdapter/issues/79

> This has made impossible to develop on iOS 15 since XCode 12 does not accept iOS 15 devices as a development target.

**Comment by LucasHimelfarb** [comment]
Hi @wildseansy sorry to bother you but could you indicate what are the steps to follow to solve the error? Reading and replicating what was said in the thread you shared, I can't solve the problem.

**Comment by wildseansy** [comment]
1. Install `patch-package`
2. 
> Hi @wildseansy sorry to bother you but could you indicate what are the steps to follow to solve the error? Reading and replicating what was said in the thread you shared, I can't solve the problem.

**Comment by wildseansy** [comment]
1. Install `patch-package` - `npm i patch-package --save-dev`
2. Add this under patches `patches/react-native-ble-plx+2.0.2.patch` 
```diff
diff --git a/node_modules/react-native-ble-plx/react-native-ble-plx.podspec b/node_modules/react-native-ble-plx/react-native-ble-plx.podspec
index def210f..0a92dcc 100644
--- a/node_modules/react-native-ble-plx/react-native-ble-plx.podspec
+++ b/node_modules/react-native-ble-plx/react-native-ble-plx.podspec
@@ -17,5 +17,5 @@ Pod::Spec.new do |s|
   s.compiler_flags = '-DMULTIPLATFORM_BLE_ADAPTER'

   s.dependency 'React'
-  s.dependency 'MultiplatformBleAdapter', '0.1.7'
+  s.dependency 'MultiplatformBleAdapter', '0.1.9'
 end
```
3. Add `yarn patch-package` to your "postinstall" script in your `package.json`
> Hi @wildseansy sorry to bother you but could you indicate what are the steps to follow to solve the error? Reading and replicating what was said in the thread you shared, I can't solve the problem.

**Comment by diogoviannaaraujo** [comment]
@wildseansy I've been trying doing the patch on postinstall but when running `pod install` I keep getting this error:

```
[!] CocoaPods could not find compatible versions for pod "MultiplatformBleAdapter":
  In Podfile:
    react-native-ble-plx (from `../node_modules/react-native-ble-plx`) was resolved to 2.0.2, which depends on
      MultiplatformBleAdapter (= 0.1.9)

None of your spec sources contain a spec satisfying the dependency: `MultiplatformBleAdapter (= 0.1.9)`.

You have either:
 * mistyped the name or version.
 * not added the source repo that hosts the Podspec to your Podfile.
```

**Comment by wildseansy** [comment]
@diogoviannaaraujo - I forgot [step 4](https://github.com/dotintent/react-native-ble-plx/issues/899#issuecomment-929490351). Update your podfile to point to new cocoapod and you'll be good to go

> 4. Add this to your Podfile

**Comment by diogoviannaaraujo** [comment]
Great, thanks that worked.

**Comment by marcoj** [comment]
And you'll probably need to
```
pod update
pod repo update
pod install
```

**Comment by LucasHimelfarb** [comment]
adding patch for podspec and also adding as pod bleAdapter is the working solution for now. Thanks for the help

**Comment by Matheuschn** [comment]
It looks like the MultiPlatformBleAdapter library has already released version 0.1.9 with the fix. If possible, can we get a new react-native-ble-plx version with these fixes? Thanks!

The way i fixed it, at least until the fixes get to Cocoapods, is by replacing the library's folder (Pods/MultiplatformBleAdapter) with a clone from the 0.1.9 release (make sure the names match). Then, run 'pod install' and it should work for now.

**Comment by LucasHimelfarb** [comment]
2.0.3 dont work for me

**Comment by mciechanowicz** [comment]
Hi everyone! The issue with Xcode 13 was fixed in the 2.0.3 version!

@LucasHimelfarb can you give more details about your Issue?

**Comment by mciechanowicz** [comment]
At this point, the problem seems to be solved so I close this issue. If anyone still has problems please report a new issue.
